zhaquirks/sonoff/swv.py

```diff
@@ -31,10 +38,87 @@ class AttributeDefs(BaseAttributeDefs):
             type=ValveState,
         )
 
+        on_time = ZCLAttributeDef(
+            id=0x5011,
+            type=t.uint16_t,
+        )
+
+    @property
+    def _is_manuf_specific(self):
+        return False
+
+
+class SwvOnOff(NoReplyMixin, CustomCluster, OnOff):
+    """SWV On Off Cluster."""
+
+    cluster_id = OnOff.cluster_id
+    cmd_values = OnOff.commands_by_name.values()
+
+    async def command(
+        self,
+        command_id: foundation.GeneralCommand | int | t.uint8_t,
+        *args,
+        manufacturer: int | t.uint16_t | None = None,
+        expect_reply: bool = True,
+        tsn: int | t.uint8_t | None = None,
+        **kwargs: typing.Any,
+    ) -> typing.Coroutine:
+        """Send a command to the device.
+
+        If on_time == 0 it keeps the normal On command (0x01)
+        If on_time != 0 it sends a on_with_timed_off (0x42), with the on_time numbers of seconds sets in the attribute
+
+        Args:
+            command_id (foundation.GeneralCommand | int | t.uint8_t): The ID of the command to send.
+            *args: Additional arguments for the command.
+            manufacturer (int | t.uint16_t | None, optional): The manufacturer ID. Defaults to None.
+            expect_reply (bool, optional): Whether to expect a reply from the device. Defaults to True.
+            tsn (int | t.uint8_t | None, optional): The transaction sequence number. Defaults to None.
+            **kwargs (typing.Any): Additional keyword arguments for the command.
+
+        Returns:
+            typing.Coroutine: A coroutine that sends the command and waits for the response.
+
+        """
+        command = self.server_commands[command_id]
+
+        if manufacturer is None and (
+            self._is_manuf_specific or command.is_manufacturer_specific
+        ):
+            manufacturer = self._manufacturer_id
+        if command_id == 0x01:
+            base_cluster = self.endpoint.in_clusters[CustomSonoffCluster.cluster_id]
+            on_time_value = base_cluster.get(
+                CustomSonoffCluster.AttributeDefs.on_time.id
+            )
+            if on_time_value is not None and on_time_value != 0:
+                command = self.server_commands[0x42]  # on_with_timed_off
```

Comment:
```suggestion
                command = self.ServerCommandDefs.on_with_timed_off
```

What does this line do? `command` isn't used anywhere.